spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Input.scala

```diff
@@ -36,8 +36,16 @@ import scala.reflect.ClassTag
  * @tparam T The numeric type in the criterion, usually which are [[Float]] or [[Double]]
  */
 @SerialVersionUID(- 8525406230282608924L)
-class Input[T: ClassTag]()(implicit ev: TensorNumeric[T])
+class Input[T: ClassTag](val inputShape: Array[Int])(implicit ev: TensorNumeric[T])
   extends AbstractModule[Activity, Activity, T] {
+
+  if (inputShape != null) {
+    val batchInputShape = Array(-1) ++ inputShape
+    setBatchInputShape(Tensor(data = batchInputShape, shape = Array(batchInputShape.length)))
```

Comment:
why use a tensor here?

Reply:
shape maybe list of array , nested array or single array.  Use `Activity`here just for unify the returning type. Any other idea for this? Actually, `Activity` is not quite straightforward here.